exist-core/src/main/java/org/exist/xquery/functions/fn/ExtCollection.java

```diff
@@ -41,35 +40,33 @@
 
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.ArrayList;
 import java.util.Iterator;
-import java.util.List;
+
+import static org.exist.xquery.FunctionDSL.*;
 
 /**
- * @author wolf
+ * @author <a href="mailto:adam@evolvedbinary.com">Adam Retter</a>
  */
-public class ExtCollection extends Function {
-
-    private static final Logger LOG = LogManager.getLogger(ExtCollection.class);
-
-    public final static FunctionSignature signature =
-            new FunctionSignature(
-                    new QName("collection", Function.BUILTIN_FUNCTION_NS),
-                    "Returns the documents contained in the collections specified in " +
-                            "the input sequence. " + XMLDBModule.COLLECTION_URI +
-                            " Documents contained in sub-collections are also included. If no value is supplied, the statically know documents are used, for the REST Server this could be the addressed collection.",
-                    new SequenceType[]{
-                            //Different from the official specs
-                            new FunctionParameterSequenceType("collection-uri", Type.STRING,
-                                    Cardinality.ZERO_OR_ONE, "The collection URI for which to include the documents")},
-                    new FunctionReturnSequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE,
-                            "The document nodes contained in or under the given collections"),
-                    true);
+public class ExtCollection extends BasicFunction {
+
+    private static final String FS_COLLECTION_NAME = "collection";
+    static final FunctionSignature[] FS_COLLECTION = functionSignatures(
+            new QName(FS_COLLECTION_NAME, Function.BUILTIN_FUNCTION_NS),
+            "Returns the documents contained in the Collections specified in the input sequence. "
+                    + XMLDBModule.COLLECTION_URI + " Documents contained in sub-collections are also included. "
+                    + "If no value is supplied, the statically know documents are used; for the REST Server this could be the collection in the URI path.",
+            returnsOptMany(Type.ITEM, "The items indicated by the Collection URI"),
+            arities(
+                    arity(),
+                    arity(
+                            optParam("arg", Type.STRING,"The Collection URI")
```

Comment:
Then name of the argument was changed from a descriptive name `collection-uri` to a generic one `arg`.
It is the name that is used in XQ31 functions and operators spec, but since
the upcoming XQ40 functions and operators spec now calls it `uri` I would like to see us use a more descriptive term and maybe settle for `uri`. Source https://qt4cg.org/specifications/xpath-functions-40/Overview.html#func-collection

Reply:
eXist-db is not yet doing anything related to XQ40.
eXist-db targets the XQuery 3.1 currently.
The goal of this PR (as per its title) is to make this `fn:collection` compliant with the XQuery 3.1 specification, the XQuery F&O 3.1 spec names this parameter `arg` and so if we are to follow the spec, then we must do the same!

Reply:
We gain compliance with the XQuery 3.1 specification!